Automattic/jetpack
# linting: Reorganize tests in lint-project-structure.sh

#### Changes proposed in this Pull Request:

Some tests may raise errors about missing composer.json because they
weren't properly guarded. Let's rearrange things so the "composer.json
exists" skips all following tests.

Also, if a line number couldn't be found for autoloader-suffix, it would
exit the script rather than continuing to print the error with no line
number and running further tests.

The check for renovate's ignore list also shouldn't end the script early if it fails.

Note the diff is probably best reviewed [with whitespace changes hidden](https://github.com/Automattic/jetpack/pull/20155/files?w=1).

#### Jetpack product discussion


None.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Is CI happy?
* `mkdir projects/packages/foobar/` then run the script. It should report the missing composer.json as an error, but should not produce any extraneous errors like "Could not open file projects/packages/foobar/composer.json".
* `echo '{ "name": "automattic/jetpack-foobar", "config": { "autoloader-suffix": "" } }' > projects/packages/foobar/composer.json`, then run the script. The output should end with debug about about "Checking .github/versions.sh vs package.json engines".
